README.md
This project is built to serve a cricket team of Mississauga, ON, Canada. This is a firebase web hosting application, along with database rules and cloud functions. In order to run this application in development mode, follow the steps below. 

1. Run this command `git clone https://github.com/ximxim/mississauga-dolphins-firebase.git`
2. Navigate to cloned project folder `cd mississiauga-dolphins-firebase`
3. Install dependencies `npm install`
4. Create an empty file in the src folder `touch src/env.production.js`
5. Start the project `npm start`

If you have any questions or inquiries, please feel free to contact me. Thanks.


CHANGELOG.md
# Change Log

## [v3.0.0](https://github.com/ximxim/mississauga-dolphins-firebase/tree/v3.0.0) (2019-07-04)
**Closed issues:**

- Add an option to score based on players [\#78](https://github.com/ximxim/mississauga-dolphins-firebase/issues/78)
- Create modal form to submit player's statistics [\#77](https://github.com/ximxim/mississauga-dolphins-firebase/issues/77)
- Add helper option in player sidebar [\#76](https://github.com/ximxim/mississauga-dolphins-firebase/issues/76)
- Create delete player option [\#75](https://github.com/ximxim/mississauga-dolphins-firebase/issues/75)
- Create add new player modal [\#74](https://github.com/ximxim/mississauga-dolphins-firebase/issues/74)
- Create players information edit modal [\#73](https://github.com/ximxim/mississauga-dolphins-firebase/issues/73)
- Create player details screen [\#72](https://github.com/ximxim/mississauga-dolphins-firebase/issues/72)
- Create Player card item for the sub sidebar [\#71](https://github.com/ximxim/mississauga-dolphins-firebase/issues/71)
- Create players menu [\#70](https://github.com/ximxim/mississauga-dolphins-firebase/issues/70)
- Add delete player option back to game [\#68](https://github.com/ximxim/mississauga-dolphins-firebase/issues/68)
- Figure out a way to scroll down to the point where the game exists [\#67](https://github.com/ximxim/mississauga-dolphins-firebase/issues/67)
- don't alter the search bar if someone goes directly to the game link [\#66](https://github.com/ximxim/mississauga-dolphins-firebase/issues/66)
- Search bar should have a clear button [\#65](https://github.com/ximxim/mississauga-dolphins-firebase/issues/65)
- Description and Image upload field should show form validation errors [\#64](https://github.com/ximxim/mississauga-dolphins-firebase/issues/64)
- Refactor Uploader to use local storage for temporary file [\#63](https://github.com/ximxim/mississauga-dolphins-firebase/issues/63)
- Move game details screen inside the sub content area [\#61](https://github.com/ximxim/mississauga-dolphins-firebase/issues/61)
- Design subsidebar [\#60](https://github.com/ximxim/mississauga-dolphins-firebase/issues/60)
- Design sidebar [\#59](https://github.com/ximxim/mississauga-dolphins-firebase/issues/59)
- Design header [\#58](https://github.com/ximxim/mississauga-dolphins-firebase/issues/58)
- start using new debugger package [\#57](https://github.com/ximxim/mississauga-dolphins-firebase/issues/57)
- Show key players information [\#56](https://github.com/ximxim/mississauga-dolphins-firebase/issues/56)
- Check for push notifications permissions on settings screen [\#55](https://github.com/ximxim/mississauga-dolphins-firebase/issues/55)
- Update firebase functions [\#53](https://github.com/ximxim/mississauga-dolphins-firebase/issues/53)
- Notify the user who updates scores, how many devices received the notification [\#51](https://github.com/ximxim/mississauga-dolphins-firebase/issues/51)
- Implement Cloud Functions Dev instance [\#50](https://github.com/ximxim/mississauga-dolphins-firebase/issues/50)
- Create duplicate library of firebase requests with dev post fix [\#49](https://github.com/ximxim/mississauga-dolphins-firebase/issues/49)
- Create duplicates of all the functions with dev postfix [\#48](https://github.com/ximxim/mississauga-dolphins-firebase/issues/48)
- Simplify index file [\#47](https://github.com/ximxim/mississauga-dolphins-firebase/issues/47)
- Test sending pus notification to test device [\#44](https://github.com/ximxim/mississauga-dolphins-firebase/issues/44)
- Manage Events [\#43](https://github.com/ximxim/mississauga-dolphins-firebase/issues/43)
- Manage Sponsors [\#42](https://github.com/ximxim/mississauga-dolphins-firebase/issues/42)
- Archive game after finished functionality [\#41](https://github.com/ximxim/mississauga-dolphins-firebase/issues/41)
- Start multiple game scores functionality [\#40](https://github.com/ximxim/mississauga-dolphins-firebase/issues/40)
- Admin login through facebook [\#39](https://github.com/ximxim/mississauga-dolphins-firebase/issues/39)
- Buy admin template [\#38](https://github.com/ximxim/mississauga-dolphins-firebase/issues/38)
- Don't pull retweets and responses [\#37](https://github.com/ximxim/mississauga-dolphins-firebase/issues/37)
- Manage numbers through Google Sheet [\#36](https://github.com/ximxim/mississauga-dolphins-firebase/issues/36)
- Manage contact details through Google Sheet [\#35](https://github.com/ximxim/mississauga-dolphins-firebase/issues/35)
- Manage teams through Google Sheet [\#34](https://github.com/ximxim/mississauga-dolphins-firebase/issues/34)
- Manage players through Google Sheet [\#33](https://github.com/ximxim/mississauga-dolphins-firebase/issues/33)
- Style slack items for instagram posts [\#31](https://github.com/ximxim/mississauga-dolphins-firebase/issues/31)
- Style slack items for twitter posts [\#30](https://github.com/ximxim/mississauga-dolphins-firebase/issues/30)
- Style slack items for facebook posts [\#29](https://github.com/ximxim/mississauga-dolphins-firebase/issues/29)
- Slack messages about recently added items should provide options to click in order to hide or show an item [\#28](https://github.com/ximxim/mississauga-dolphins-firebase/issues/28)
- Firebase function should only be triggered by users in admin channel [\#26](https://github.com/ximxim/mississauga-dolphins-firebase/issues/26)
- Integrate newsfeed function trigger in slack [\#25](https://github.com/ximxim/mississauga-dolphins-firebase/issues/25)
- Pull newsFeed data from firebase to compare with new batch [\#24](https://github.com/ximxim/mississauga-dolphins-firebase/issues/24)
- Use NGROK if possible to open a static port [\#23](https://github.com/ximxim/mississauga-dolphins-firebase/issues/23)
- Slack feature to hide newsfeed items [\#22](https://github.com/ximxim/mississauga-dolphins-firebase/issues/22)
- Slack command to fetch news feed items [\#21](https://github.com/ximxim/mississauga-dolphins-firebase/issues/21)

src/env.develop.js
export default {
    firebase: {
        apiKey: 'AIzaSyCPu1cq_G_tzSvrnmKMV02g8yES4j_5kIg',
        authDomain: 'mississauga-dolphins.firebaseapp.com',
        databaseURL: 'https://mississauga-dolphins-dev.firebaseio.com',
        projectId: 'mississauga-dolphins',
        storageBucket: 'mississauga-dolphins.appspot.com',
        messagingSenderId: '562485347443',
    },
    newGame: {
        home: {
            name: 'Pakistan',
            score: 200,
            wickets: 5,
            overs: 4,
            batting: true,
        },
        visitor: {
            name: 'India',
            score: 100,
            wickets: 8,
            overs: 9,
            batting: false,
        },
        striker: 'John Smith',
        nonStriker: 'Azim Ahmed',
        bowler: 'Adeel Ahmed',
    },
    newGameEvent: {
        cover: {
            source: 'https://firebasestorage.googleapis.com/v0/b/mississauga-dolphins.appspot.com/o/events%2F452x47.jpg?alt=media&token=8fe48f25-c251-417a-96b1-4c44ef86f0e8',
            thumbnail: 'https://firebasestorage.googleapis.com/v0/b/mississauga-dolphins.appspot.com/o/events%2F452x47.jpg_thumbnail?alt=media&token=5eef048f-6819-4177-811b-280b3cd2b45e',
        },
        description: 'Match No: 452\nDivision: 1st Division - CN Pro\nRound type: Preliminary',
        division: '100',
        game: true,
        match_no: '452',
        place: {
            name: 'Hole in the groun',
        },
        players: [2, 3, 75, 47, 161, 153, 33, 167, 149, 140, 114, 1, 8, 7, 61, 99],
        round_type: 'Preliminary',
        start_time: '2018-09-09T19:00:00.000Z',
        title: 'Custom Dev Game',
    },
    newPlayer: {
        '3FORs': 1,
        '4FORs': 2,
        '5FORs': 3,
        '6FORs': 4,
        '7FORs': 5,
        AVG_RUN: 6,
        BATING_STYLE: 'Right Handed',
        BATTING: {
            '100s': 8,
            '4s': 9,
            '50s': 10,
            '6s': 11,
            AVG: 12,
            HIGHEST: 13,
            INNINGS: 14,
            NO: 15,
            RUNS: 16,
        },
        BOWLING: {
            MDNS: 17,
            OVERS: 18,
            WKTS: 19,
        },
        BOWLING_STYLE: 'Medium Fast',
        CTS: 21,
        DATE_JOINED: '2019',
        DOB: '1987',
        DP: '',
        ECON: 22,
        FIRST_NAME: 'Azim',
        FIRST_PLAYED: 'Ahmed',
        GENERAL: {
            MTCHS: 23,
        },
        GIVEN: 'Xim',
        HONOR: 'Something he did was good',
        LAST_NAME: 'Ahmed',
        LAST_PLAYED: '2019',
        ROLE: 'Bowler',
        SR_OVER: 24,
        ST: 25,
        WK_CTS: 26,
        id: null,
        avatar: {
            source: '',
            thumbnail: '',
        },
    },
    newSponsor: {
        ID: null,
        IMAGE: '',
        NAME: 'Google',
        TAG_LINE: 'Amazing company to work for',
        WEBSITE: 'https://google.ca',
    },
    login: {
        username: 'azim.ahmed7@gmail.com',
        password: '826474',
    },
};
